### ISSUE TITLE
Loss is increasing when fine-tuning from a Megatron-Deepspeed pretrained checkpoint.

### ISSUE THREAD

**Issue by SefaZeng** [microsoft/Megatron-DeepSpeed#358]
Thanks for this framework. I have pre-train a model using  Megatron-Deepspeed, and I want to use more data to continue training. But I found the loss of continued training is oscillating and increasing
<img width="575" alt="continue_train" src="https://github.com/microsoft/Megatron-DeepSpeed/assets/13918630/be513d66-9492-41e8-83a4-3e180299f77e">
There are some spikes with the loss increasing and the fine-tuned model can not use fp16 for inference as it is trained with BF16 in both the pretraining stage and continue training stage.

The script is like:
```
ds_args=""
ds_args=" --deepspeed ${ds_args}"
ds_args=" --deepspeed_config=$DS_CONFIG ${ds_args}"
ds_args=" --zero-stage=$ZERO_STAGE ${ds_args}"
ds_args=" --deepspeed-activation-checkpointing ${ds_args}"
 

DISTRIBUTED_ARGS="--nproc_per_node $GPUS_PER_NODE --nnodes $NNODES --node_rank $NODE_RANK --master_addr $MASTER_ADDR --master_port $MASTER_PORT"
 
torchrun $DISTRIBUTED_ARGS \
       ${CODE_PATH}/pretrain_gpt.py \
       --tensor-model-parallel-size $TP \
       --pipeline-model-parallel-size $PP \
       --no-pipeline-parallel \
       --num-layers $NUM_LAYERS \
       --hidden-size $HIDDEN_SIZE \
       --ffn-hidden-size $FFN_HIDDEN_SIZE \
       --num-attention-heads $NUM_HEADS \
       --micro-batch-size $MICRO_BATCH_SIZE \
       --global-batch-size $GLOBAL_BATCH_SIZE \
       --seq-length $SEQ_LENGTH \
       --max-position-embeddings $SEQ_LENGTH \
       --train-iters $TRAIN_STEPS \
       --save $CHECKPOINT_PATH \
       --load $PRE_MODEL \
       --train-data-path $DATASET \
       --valid-data-path $DATASET_VALID \
       --data-impl mmap \
       --tokenizer-type HFTokenizer \
       --tokenizer-model $TOKENIZER_PATH \
       --distributed-backend nccl \
       --lr $LR \
       --lr-decay-style cosine \
       --min-lr $MIN_LR \
       --weight-decay $WEIGHT_DECAY \
       --clip-grad $GRAD_CLIP \
       --lr-warmup-iters $LR_WARMUP_STEPS \
       --optimizer adam \
       --adam-beta1 0.9 \
       --adam-beta2 0.95 \
       --log-interval 1 \ 
       --save-interval 1000 \
       --eval-interval 1000000000 \
       --eval-iters 1000 \
       --use-rotary-position-embeddings \
       --num-key-value-heads $NUM_KV_HEADS \
       --no-query-key-layer-scaling \
       --attention-dropout 0 \ 
       --hidden-dropout 0 \ 
       --swiglu \
       --normalization rmsnorm \
       --disable-bias-linear \
       --use-flash-attn \
       --bf16 \
       $ds_args \
       --finetune \
       --no-load-optim \
       --no-load-rng \
       --no-load-lr-state \
       --no-initialization \
```
Any tips for this?
